pkg/frontend/querymiddleware/astmapper/instant_splitting_test.go

```diff
@@ -297,7 +298,12 @@ func concatOffsets(splitInterval time.Duration, offsets int, queryTemplate strin
 	offsetIndex := offsets
 	for offset := range queries {
 		offsetIndex--
-		offsetQuery := fmt.Sprintf("[%s]%s", splitInterval, getSplitOffset(splitInterval, offsetIndex))
+		offsetSplitInterval := splitInterval
+		if offset > 0 {
+			offsetSplitInterval -= time.Millisecond
```

Comment:
Just a curiosity 🤔 why is the unit `time.Millisecond` and not for instance `time.Nanosecond`? Is this the same unit used in the sharding?

Reply:
Millisecond is the precision unit of our database.